Microsoft.Identity.Web/Client/TokenCacheProviders/InMemory/MSALPerUserMemoryTokenCacheProvider.cs

```diff
@@ -94,11 +94,7 @@ private void UserTokenCacheAfterAccessNotification(TokenCacheNotificationArgs ar
             // if the access operation resulted in a cache update
             if (args.HasStateChanged)
             {
-                string cacheKey = args.Account?.HomeAccountId?.Identifier;
-                if (string.IsNullOrEmpty(cacheKey))
-                {
-                    cacheKey = httpContextAccessor.HttpContext.User.GetMsalAccountId();
-                }
+                string cacheKey = httpContextAccessor.HttpContext.User.GetMsalAccountId();
```

Comment:
This will ensure that the key is made of the oid.tid of the guest in the case of a guest account (not the home account id identifier).

Reply:
@MarkZuber @bgavrilMS is it ok for the token cache providers to use their key as oid.tid? 
I ask that because a few weeks ago, one of you two mentioned that we should use the account identifier as the key, which is uid.utid, and that is when the code `string cacheKey = args.Account?.HomeAccountId?.Identifier` got introduced.